lib/matplotlib/backend_tools.py

```diff
@@ -0,0 +1,991 @@
+
+    def _add_tool_cbk(self, event):
+        """Process every newly added tool"""
+        if event.tool is self:
+            return
+
+        self._add_tool(event.tool)
+
+    def _set_cursor_cbk(self, event):
+        if not event:
+            return
+
+        if not getattr(event, 'inaxes', False) or not self._cursor:
+            if self._last_cursor != self._default_cursor:
+                self.set_cursor(self._default_cursor)
+                self._last_cursor = self._default_cursor
+        elif self._cursor:
+            cursor = self._cursor
+            if cursor and self._last_cursor != cursor:
+                self.set_cursor(cursor)
+                self._last_cursor = cursor
+
+    def set_cursor(self, cursor):
+        """
+        Set the cursor
+
+        This method has to be implemented per backend
+        """
+        raise NotImplementedError
+
+
+class ToolCursorPosition(ToolBase):
+    """
+    Send message with the current pointer position
+
+    This tool runs in the background reporting the position of the cursor
+    """
+    def __init__(self, *args, **kwargs):
+        ToolBase.__init__(self, *args, **kwargs)
+        self._idDrag = self.figure.canvas.mpl_connect(
+            'motion_notify_event', self.send_message)
+
+    def send_message(self, event):
+        """Call `matplotlib.backend_managers.ToolManager.message_event`"""
+        if self.toolmanager.messagelock.locked():
+            return
+
+        message = ' '
+
+        if event.inaxes and event.inaxes.get_navigate():
+            try:
+                s = event.inaxes.format_coord(event.xdata, event.ydata)
+            except (ValueError, OverflowError):
+                pass
+            else:
+                message = s
+        self.toolmanager.message_event(message, self)
+
+
+class RubberbandBase(ToolBase):
+    """Draw and remove rubberband"""
+    def trigger(self, sender, event, data):
+        """Call `draw_rubberband` or `remove_rubberband` based on data"""
+        if not self.figure.canvas.widgetlock.available(sender):
+            return
+        if data is not None:
+            self.draw_rubberband(*data)
+        else:
+            self.remove_rubberband()
+
+    def draw_rubberband(self, *data):
+        """
+        Draw rubberband
+
+        This method must get implemented per backend
+        """
+        raise NotImplementedError
+
+    def remove_rubberband(self):
+        """
+        Remove rubberband
+
+        This method should get implemented per backend
+        """
+        pass
+
+
+class ToolQuit(ToolBase):
+    """Tool to call the figure manager destroy method"""
+
+    description = 'Quit the figure'
+    default_keymap = rcParams['keymap.quit']
+
+    def trigger(self, sender, event, data=None):
+        Gcf.destroy_fig(self.figure)
+
+
+class ToolEnableAllNavigation(ToolBase):
+    """Tool to enable all axes for toolmanager interaction"""
+
+    description = 'Enables all axes toolmanager'
```

Comment:
Yes, I didn't want to remove old functionality, so I left it there as it was. 
A smaller PR (after merge) correcting that would be cleaner.